### DISCUSSION TITLE
ID Tags

### DISCUSSION THREAD

**Discussion by ats3788** [pschatzmann/arduino-audiokit#113] [Q&A] [answered]
Hallo Danke 
Hr. Schatzmann für ihre tollen Sketche.
Leider bekomme ich die Beispiele vom Kollegen Schreibfaul nicht zum laufen.
Gibt es eine Möglichkeit die ID Tags auszulesen
So wie vom Kollegen Schreibfaul
// optional
void audio_info(const char *info){
    Serial.print("info        "); Serial.println(info);
}
void audio_id3data(const char *info){  //id3 metadata
    Serial.print("id3data     ");Serial.println(info);
}
void audio_eof_mp3(const char *info){  //end of file
    Serial.print("eof_mp3     ");Serial.println(info);
}
void audio_showstation(const char *info){
    Serial.print("station     ");Serial.println(info);
}
void audio_showstreaminfo(const char *info){
    Serial.print("streaminfo  ");Serial.println(info);
}
void audio_showstreamtitle(const char *info){
    Serial.print("streamtitle ");Serial.println(info);
}
void audio_bitrate(const char *info){
    Serial.print("bitrate     ");Serial.println(info);
}
void audio_commercial(const char *info){  //duration in sec
    Serial.print("commercial  ");Serial.println(info);
}
void audio_icyurl(const char *info){  //homepage
    Serial.print("icyurl      ");Serial.println(info);
}
void audio_lasthost(const char *info){  //stream URL played
    Serial.print("lasthost    ");Serial.println(info);
}
void audio_eof_speech(const char *info){
    Serial.print("eof_speech  ");Serial.println(info);
}

**Comment by pschatzmann** [comment, marked as answer]
Not as part of this project (which implements only a device driver for the AudioTools). 
Metdata has been implemented in the [AudioTools](https://github.com/pschatzmann/arduino-audio-tools): I am supporting ID3 and ICY metadata!

You forgot to mention in what scenario you want to use this:
- You can register a callback in the AudioPlayer
- You can register a callback in the ICYStream
- You can use the MetaDataOutput as output destination
